pkg/vault/client.go

```diff
@@ -197,16 +193,44 @@ func WriteSecret(instanceAddr string, secretPath string, secretData map[string]i
 	}
 }
 
-// read secret from vault
-func ReadSecret(instanceAddr string, secretPath string) *api.Secret {
-	secret, err := getClient(instanceAddr).Logical().Read(secretPath)
+// read secret from vault and return the secret map
+func ReadSecret(instanceAddr, secretPath, engineVersion string) map[string]interface{} {
+	raw, err := getClient(instanceAddr).Logical().Read(secretPath)
 	if err != nil {
 		log.WithError(err).WithFields(log.Fields{
-			"path":     secretPath,
-			"instance": instanceAddr,
+			"path":          secretPath,
+			"instance":      instanceAddr,
+			"engineVersion": engineVersion,
 		}).Fatal("[Vault Client] failed to read Vault secret")
 	}
-	return secret
+	if raw == nil {
+		return nil
+	}
+	// vault api returns different payload depending on version
+	switch engineVersion {
+	case KV_V1:
+		return raw.Data
+	case KV_V2:
+		if len(raw.Data) == 0 {
```

Comment:
I'm sure there is a reason why this is used here and not in KV V1

Reply:
using a length check to avoid keyerror on next portion `raw.Data["data"].(map[string]interface{})`

I suppose this could be omitted as it'd still be caught by the attempted type assertion